derek_love/OOP/bank.rb

```diff
@@ -0,0 +1,62 @@
+class Bank
+    attr_accessor :checking, :savings
+
+    @@num_of_accts = 0
+    @@interest_rate = 0.01
+
+    def initialize
+        @checking = 0
+        @savings = 0
+
+        @@num_of_accts += 1
+        p ("Accounts Created: #{@@num_of_accts}")
+    end
+    
+    def acct_info
+        p "Account Number: #{bank_acct}"
+        p "Account Totals: #{total}"
+        p "Checking Total: #{checking_acct}"
+        p "Savings Total: #{savings_acct}"
+        p "Interest Rate: #{@@interest_rate}"
+    end
+    
+    def checking_acct
+        return @checking
+        self
```

Comment:
With the return in these methods, the self statement won't be executed. So essentially the self here is dead code.